gamearea.py

```diff
@@ -356,12 +356,20 @@ def __draw_end_message(self, context):
                 catsCount[cat.cat_id-1]+=1
 
             for cat in self.cats:
-                if cat.x < alloc.width // 2 and catsCount[cat.cat_id-1] % 2 != 0:
-                    correctly_placed = False
-                    break
-                if cat.x > alloc.width // 2 and catsCount[cat.cat_id-1] % 2 == 0:
-                    correctly_placed = False
-                    break
+                if self.sides[0] == SideType.ODD:
+                    if cat.x < alloc.width // 2 and catsCount[cat.cat_id-1] % 2 == 0 :
+                        correctly_placed = False
+                        break
+                    if cat.x > alloc.width // 2 and catsCount[cat.cat_id-1] % 2 != 0:
+                        correctly_placed = False
+                        break
+                else:
+                    if cat.x < alloc.width // 2 and catsCount[cat.cat_id-1] % 2 != 0:
+                        correctly_placed = False
+                        break
+                    if cat.x > alloc.width // 2 and catsCount[cat.cat_id-1] % 2 == 0:
+                        correctly_placed = False
+                        break
```

Comment:
`correctly_placed` is never true in your checks so the placement would always be wrong regardless, anything `% 2 == 0` should be even, not odd.

Did you test this?

Reply:
I did test this, If you look at `line 352`, Correctly_placed is declared as true, now if the user places cats incorrectly in any of the possible cases it is making it false.

In that divided game, sometimes odd count cats are to be placed at right or some other time it has to be placed at left.

Before it was only checked for the right side. 
 
`if self.sides[0] == SideType.ODD` tells whether the odd message is on the right or left, and then accordingly it's checking.

Reply:
I can see that `correctly_placed` was declared as True earlier, I was referring to your checks.

`correctly_placed` starts out as True and gets assigned False regardless of the outcome and both checks are different but both assign `correctly_placed` to False and that's not right as both checks shouldn't equate to False, anything `% 2 == 0` equates to that thing being even and in your else statement it equates to False.

The multiple checks isn't necessary as something is even when `% 2 == 0` is True and False otherwise.

Reply:
Um so I think you are getting something wrong here.
First we are checking this `if self.sides[0] == SideType.ODD:`, so through this we will know if odd message is on right or left, now let's say it's `true`,

then, 
1. `if cat.x < alloc.width // 2 and catsCount[cat.cat_id-1] % 2 == 0 :
                        correctly_placed = False
                        break
`
this if condition will check if the cats that are **placed at left has a count of even or not**, and if any of those cat has a even count we are making correctly_placed as false, as cats of *odd* count should be present there, and it is breaking out of the for loop as next conditions are unnecessary to be checked.

now the next if condition
2. `if cat.x > alloc.width // 2 and catsCount[cat.cat_id-1] % 2 != 0:
                        correctly_placed = False
                        break
`
checks if any cats that are **placed at right is odd,** and if so correclty_placed is made false, as even count cats should be present there.

The need for both of the if condition is that, if i remove 2nd if condition and I place all the cats to the right, it will make it as a win, regardless of the count.

the else condition is for when the odd message is on left and it just switches both of these cases for right and left.

Reply:
![Screencast from 2023-06-23 15-17-10](https://github.com/sugarlabs/classify-cats/assets/16203834/ac8e0510-17f0-4f94-bbc6-f68659c16482)

This is the outcome of your change.

@quozl can you test?

Reply:
Isn't this what it should be expected?

The green cat has an odd count, and placing one of its cats on the even side is a loss.

I guess we are deciphering the question differently.

`classify each kind of cat as even and odd`
since **each kind** was specified, the cats having odd count, count being counted on basis of color and not the cats placed on any section.

Reply:
If you place all the Green, yellow and blue cats on left, as they have an odd count of 3, 3 and 1 respectively and red cat on right as it's count is 2 which is even you will have a win. 

This is what i thought the gameplay was.

Reply:
Yes you're right, I got confused and it's been a while I ran this activity.

Reply:
We should add a help to the activity.